foundationdb/tests/timekeeper.rs

```diff
@@ -0,0 +1,78 @@
+use foundationdb::timekeeper::{hint_version_from_timestamp, HintMode};
+use std::cmp::{max, min};
+use std::time::SystemTime;
+
+#[tokio::test]
+async fn timekeeper() {
+    let _guard = unsafe { foundationdb::boot() };
+    let database = foundationdb::Database::new_compat(None)
+        .await
+        .expect("Unable to create database");
+    let trx_read = database.create_trx().expect("Unable to create transaction");
+    let now = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .expect("Unable to get timestamp")
+        .as_secs();
+    let read_version = trx_read
+        .get_read_version()
+        .await
+        .expect("Unable to get read version") as u64;
+    // A new transaction is needed because the one which get the read version has no
+    // knowledge about the new System Namespace Keyspace thus, further timekeeper
+    // keys are unknown for the first transaction.
+    // Creating a new one, make the first to be commited. The second will have the right
+    // timekeeper state.
+    let trx = database.create_trx().expect("Unable to create transaction");
+    tokio::time::sleep(std::time::Duration::from_secs(10)).await;
```

Comment:
suggestion: I don't think that creating a txn then sleeping is the right order of things. We should sleep, then make the transaction